### DISCUSSION TITLE
Error Could not save cache , no writable directory

### DISCUSSION THREAD

**Discussion by mvtkurd** [bitcraze/discussions#68] [Q&A] [answered]
Hi @krichardsson 
I'm trying to flash ai deck esp but I got this error : Error Could not save cache , no writable directory
I'm using VM and docker and also I attached this image to make it more clear .

![Screenshot 2022-05-07 201806](https://user-images.githubusercontent.com/32674939/167266856-772fa13a-8432-4ac0-8c4e-777cecd34587.jpg)

**Comment by knmcguire** [comment]
Hi! We have updated the [aideck tutorial](https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/) according to the latest release. Could you go through that and try again?

**Comment by mvtkurd** [comment]
Hi @knmcguire 

I've followed the tutorial and this is the output of cfclient console:
As you see, there's no `ESP32: I (910) SYS: Initialized` in console but the LED1 blink with frequency 2Hz.
And AIDECK-ROUTER : Not setting up WIFI !


![second](https://user-images.githubusercontent.com/32674939/171211825-a0626571-0b8d-49f8-9577-6a537b124cd0.jpg)

**Comment by evoggy** [comment]
I think that maybe the VM is causing issues for this. Could you please try to update the STM32 firmware of the crazyflie using the cfloader to see if this works?

**Comment by mvtkurd** [comment]
Hi @evoggy ,

I cloned the latest release of firmware (2022.05) and the firmware has been built and flashed correctly by radio dongle but now how can I check the wifi has been set up ?

![4](https://user-images.githubusercontent.com/32674939/171378585-9a4f499e-a64c-4301-a61f-f82ab094ff0a.jpg)

**Comment by knmcguire** [comment]
I'm converting this to discussions as this is not yet a bug report yet, so you can find it here: discussions.bitcraze.io

**Comment by mvtkurd** [comment]
@knmcguire thank you for opening this discussion. I thought the problem would be related to esp firmware but I updated also the esp firmware to the latest release  and still doesn't see the `ESP32: I (910) SYS: Initialized` in the console !!

**Comment by mvtkurd** [comment]
@evoggy @knmcguire 

I found the solution by updating ESP firmware by jtag first and then update STM32 firmware of the crazyflie using the cfloader. Now you see the console output in attached image but there's also a warning `Gap8 :  Binary application header doesn't seem ok , not exiting bootloader`
what does it mean ?

![solve](https://user-images.githubusercontent.com/32674939/171430399-46e9ac1e-9ffa-4083-a0cd-a200f2c19485.jpg)

**Reply by maichongju** [reply]
For me is a similar story. When I try to flash, the program quit after `Could not save cache, no writable directory`. But without any message said it fail. 

Screen Shot attached:
![image](https://user-images.githubusercontent.com/31859452/171700072-fad113d3-10f9-48cc-88ae-f64a67aa0a83.png)

**Reply by evoggy** [reply]
`Gap8 : Binary application header doesn't seem ok , not exiting bootloader` mean that there's no application flashed on the GAP8 (except for the bootloader). If you try to flash the application does this go away?

For the flashing that's not working, please remove the address at the end (i.e E7E7 etc)

**Reply by mvtkurd** [reply]
@evoggy 
I tried to flash `aideck_gap8_wifi_img_streamer_with_ap.bin` file by cfloader but I got this error :

![wifi](https://user-images.githubusercontent.com/32674939/172022833-4f53a162-c2bd-4ff7-927c-acdd14cfa1c6.jpg)

**Reply by krichardsson** [reply]
It looks like your URI is wrong, it should be something like `radio://0/80/2M`

**Reply by mvtkurd** [reply]
@krichardsson I've also checked with `-w ://0/80/1M` but the same error occurred. But then I test the Wi-Fi streaming example and it works !

**Reply by evoggy** [reply]
You need to add radio in front of the URI, like this ```-w radio://0/80/1M```. Does it work when adding this?

**Comment by knmcguire** [comment]
I'm starting a new thread here to keep it managable.

@mvtkurd

The standard uploaded binary is the WiFi example already on the AIdeck. So it seems that flashing is still an issue for you. 

Make sure to go through all the steps of [this tutorial again](https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/) from the start, reflashing everything, with the latest version of the cfclient and cflib. Then try to flash the wifi binary again.

**Reply by knmcguire** [reply]
Let us know here if you are still having problems

**Reply by mvtkurd** [reply]
@knmcguire 
Actually the problem was that in the tutorial is written to flash the binary use `cfloader flash aideck_gap8_wifi_img_streamer_with_ap.bin deck-bcAI:gap8-fw -w [CRAZYFLIE_URI]` and I set `CRAZYFLIE_UR `wrong😅 in this way `-w ://0/80/1M` and thanks to @evoggy 's hint, now it works.

thanks for supporting!

**Reply by evoggy** [reply]
Good to hear that it worked out!

**Reply by knmcguire** [reply]
> @knmcguire Actually the problem was that in the tutorial is written to flash the binary use `cfloader flash aideck_gap8_wifi_img_streamer_with_ap.bin deck-bcAI:gap8-fw -w [CRAZYFLIE_URI]` and I set `CRAZYFLIE_UR `wrong😅 in this way `-w ://0/80/1M` and thanks to @evoggy 's hint, now it works.
> 
> thanks for supporting!

@mvtkurd could you use the 'suggest a answer' box underneath to enter the solution to your problem like you did here and press 'mark as answer' ?  This way, the solution will be nicely placed on top right underneath your initial question and you'll get kudos😄

**Comment by mvtkurd** [comment, marked as answer]
To solve the problem related to WIFI initialization I updates the firmware in this way : 

1. updating ESP firmware by jtag  
2. then update STM32 firmware of the crazyflie using the cfloader

And to solve the issue related to flashing binary file `aideck_gap8_wifi_img_streamer_with_ap.bin` we have to set CRAZYFLIE_URI in this way : `-w radio://0/80/1M`.
